storm-core/src/jvm/org/apache/storm/daemon/supervisor/Container.java

```diff
@@ -0,0 +1,493 @@
+            String supervisorId, ResourceIsolationInterface resourceIsolationManager) throws IOException {
+        this(AdvancedFSOps.make(conf), port, assignment, conf, null, supervisorId, resourceIsolationManager);
+    }
+    
+    /**
+     * Constructor to use when trying to recover a container from just the worker ID.
+     * @param workerId the id of the worker
+     * @param conf the config of the supervisor
+     * @param supervisorId the id of the supervisor
+     * @param resourceIsolationManager the isolation manager.
+     * @throws IOException on any error
+     */
+    protected Container(String workerId, Map<String, Object> conf, 
+            String supervisorId, ResourceIsolationInterface resourceIsolationManager) throws IOException {
+        this(AdvancedFSOps.make(conf), -1, null, conf, null, supervisorId, resourceIsolationManager);
+    }
+    
+    /**
+     * Kill a given process
+     * @param pid the id of the process to kill
+     * @throws IOException
+     */
+    protected void kill(long pid) throws IOException {
+        Utils.killProcessWithSigTerm(String.valueOf(pid));
+    }
+    
+    /**
+     * Kill a given process
+     * @param pid the id of the process to kill
+     * @throws IOException
+     */
+    protected void forceKill(long pid) throws IOException {
+        Utils.forceKillProcess(String.valueOf(pid));
+    }
+    
+    @Override
+    public void kill() throws IOException {
+        LOG.info("Killing {}:{}", _supervisorId, _workerId);
+        Set<Long> pids = getAllPids();
+
+        for (Long pid : pids) {
+            kill(pid);
+        }
+    }
+    
+    @Override
+    public void forceKill() throws IOException {
+        LOG.info("Force Killing {}:{}", _supervisorId, _workerId);
+        Set<Long> pids = getAllPids();
+        
+        for (Long pid : pids) {
+            forceKill(pid);
+        }
+    }
+    
+    /**
+     * Read the Heartbeat for the current container.
+     * @return the Heartbeat
+     * @throws IOException on any error
+     */
+    public LSWorkerHeartbeat readHeartbeat() throws IOException {
+        LocalState localState = ConfigUtils.workerState(_conf, _workerId);
+        LSWorkerHeartbeat hb = localState.getWorkerHeartBeat();
+        LOG.trace("{}: Reading heartbeat {}", _workerId, hb);
+        return hb;
+    }
+
+    /**
+     * Is a process alive and running?
+     * @param pid the PID of the running process
+     * @param user the user that is expected to own that process
+     * @return true if it is, else false
+     * @throws IOException on any error
+     */
+    protected boolean isProcessAlive(long pid, String user) throws IOException {
+        if (Utils.IS_ON_WINDOWS) {
+            return isWindowsProcessAlive(pid, user);
+        }
+        return isPosixProcessAlive(pid, user);
+    }
+    
+    private boolean isWindowsProcessAlive(long pid, String user) throws IOException {
+        boolean ret = false;
```

Comment:
Sorry, was a bit too fast there. tasklist also outputs user name in domain\username format if passed the /v flag.
`C:\Users\Stig>tasklist /nh /fi "pid eq 10300" /v`
outputs
`bash.exe                     10300 Console                   21        888 K Unknown         STIG-LAPTOP\Stig                                        0:00:00 N/A` 

The columns are "Image Name, PID, Session Name, Session#, Mem Usage, Status, User Name, CPU Time, Window Title"